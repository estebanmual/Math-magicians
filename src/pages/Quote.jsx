import { useState, useEffect } from 'react';
import style from '../styling/Quote.module.css';

const axios = require('axios');

function Quote() {
  const [quote, setQuote] = useState('');

  useEffect(async () => {
    await axios.get('https://random-math-quote-api.herokuapp.com/')
      .then((response) => setQuote(response.data));
  }, []);

  return (
    <div className={style.Quote}>
      <h2 className={style.Quote__quote}>{quote.quote}</h2>
      <h3 className={style.Quote__author}>{quote.author}</h3>
    </div>
  );
}

export default Quote;
