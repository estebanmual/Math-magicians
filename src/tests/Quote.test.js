import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Quote from '../pages/Quote';

const axios = require('axios');

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('API data response', async () => {
  const request = await axios.get('https://random-math-quote-api.herokuapp.com/')
    .then((response) => response.data);

  const response = (request) => ({ author: request.author, id: request.id, quote: request.quote });

  const div = document.createElement('div');
  ReactDOM.render(<Quote />, div);
  ReactDOM.unmountComponentAtNode(div);

  expect(request).toMatchObject(response(request));
});
