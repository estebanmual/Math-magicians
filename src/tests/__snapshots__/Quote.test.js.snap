// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="Quote"
>
  <h2
    className="Quote__quote"
  />
  <h3
    className="Quote__author"
  />
</div>
`;
